docs/docs/propeller/index.md

```diff
@@ -0,0 +1,40 @@
+---
+sidebar_position: 4
+---
+
+# Propeller
+
+Propeller is EquinoxJS's "[propulsion](https://github.com/jet/propulsion)
+lite". It's a set of abstractions for building reactive event processing
+pipelines catering specifically for event sourced reactions from Equinox's
+stores.
+
+## StreamsSink
+
+The `StreamsSink` receives batches of events from a source and intelligently
+manages handling them.
+
+- **Stream management**: maintains an ordered queue of incoming streams,
+  ensuring that reactions processing aligns with the ordering of the triggering
+  events; the aim being to be able to checkpoint attained progress ('tick off
+  batches') as soon as possible.
+- **Concurrent stream handling**: To ensure optimal performance, `StreamsSink`
+  restricts the number of streams that are processing at any given moment to a
+  configurable limit. Additionally guarantees that each stream can have at most
+  one handler actively processing.
+- **Batch merging**: intelligently merges new events with existing streams.
+  This allows for a more streamlined event processing, ensuring that events
+  from multiple batches related to the same stream are handled together.
+
+## TailingFeedSource
+
+`TailingFeedSource` continuously crawls a feed, submitting batches of events to
+a `Sink`.
+
+- **Crawling**: Continuously crawls a given feed forward from a checkpointed
+  position (or the start of the feed if no batches have yet completed
+  processing). It pauses momentarily between crawls once it's reached the tail
+  of the feed.
+- **Checkpointing**: Regularly updates the checkpoint position to reflect the
```

Comment:
regularly sounds like it's a pretty random sort of a contraption. Periodic?

Reply:
maybe it should allude to or mention that the wait to checkpoint is not part of the processing time overall and/or that it dedups checkpoint writes if there are lots in a replay scenario?